socketServer.cpp

```diff
@@ -201,6 +204,10 @@ void SocketServer::processBinaryMessage(QByteArray message)
                 }
             }
             clients[room_port].append(qMakePair(client, qMakePair(json.value("player_name").toString(), player_num)));
+
+            if (json.contains("input_delay")) {
+                emit inputDelayChanged(player_num - 1, json.value("input_delay").toInt());
```

Comment:
`player_num` is one-based, but the inputDelays array in udpServer is zero-based, which is why it is `player_num - 1`